src/CodeGen_X86.cpp

```diff
@@ -417,11 +377,18 @@ void CodeGen_X86::visit(const Div *op) {
 
     // Detect if it's a small int division
     const Broadcast *broadcast = op->b.as<Broadcast>();
```

Comment:
Functions that do that exist in Simplify.cpp. I'll promote them to
IROperator and use them here too.

On Tue, Oct 6, 2015 at 10:30 AM, Zalman Stern <notifications@github.com>
wrote:

> In src/CodeGen_X86.cpp
> <https://github.com/halide/Halide/pull/938#discussion_r41295363>:
>
> > @@ -417,11 +377,18 @@ void CodeGen_X86::visit(const Div *op) {
> >
> >      // Detect if it's a small int division
> >      const Broadcast *broadcast = op->b.as<Broadcast>();
>
> This code appears here and in X86 so it might be worth a helper routine.
> I'd like that just so the number of extraneous local variables in scope is
> reduced. Another approach would be to have "bool value_if_int_constant(Expr
> e, int64_t &v)" and "bool value_if_uint_constant(Expr e, uint64_t &v)"
> which do all of this. Code would be:
> int64_t int_imm = 0;
> value_if_int_constant(op->b, int_imm);
> uint64_t uint_imm = 0;
> value_if_uint_constant(op->b, int_imm);
> (Design and naming can probably be improved...)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/halide/Halide/pull/938/files#r41295363>.
>